### DISCUSSION TITLE
BITL qr code param gets overwrite

### DISCUSSION THREAD

**Discussion by maxdor1** [gopro/labs#873] [Q&A]
Is it normal that the BITL param is getting overwrite when using Quik to setup a livestream?

I set it to 8000 (!MBITL=8) but when using Quik to setup a livestream, my rtmp server receive a videobitrate of 5000.

Thank you!

**Comment by dnewman-gpsw** [comment]
Maybe.  Does the bitrate remain higher if you use Labs to start the livestream?

**Comment by maxdor1** [comment]
If I use a qr code to launch the stream, it seems to pick that last setting
that was used.

IE: launch the stream with python with open-gopro, set to 7000, the bitrate
is 7000, scan qr code with 8000, scan qr code to launch the stream, it
stays at 7000..

Launch the stream with python at 8000, it goes to 8000 than scan a qr code
for 7000, scan a qr code to launch the stream, it stays at 8000...

I dont know if im doing something wrong 😅

Le sam. 20 avr. 2024, 12 h 43, David Newman ***@***.***> a
écrit :

**Reply by dnewman-gpsw** [reply]
I haven't used Open GoPro, so I don't know how that might relate.  So when are you setting bit-rates and when are they accepted?  Are you setting livestream bit-rate via Open GoPro (I didn't know you can do that.) Seems like there is some control, just not ideal.    I do see a potential weakness in the Labs code design.  If livestream, set the bit-rate, potentially the bitrate will only be changed if set before livestreaming.

**Comment by maxdor1** [comment]
I scan the BITL qr code before going live.

Like I said, the only way that seems to apply bitrate correctly is by using
open-gopro livestream demo.



Le sam. 20 avr. 2024, 13 h 20, David Newman ***@***.***> a
écrit :

**Reply by dnewman-gpsw** [reply]
So you can set the bit-rate in open gopro?

There seems to be a potential bug in the way Labs sets the bitrate for LS, seemly only applying to the next livestream (not the current.)

**Comment by maxdor1** [comment]
In the demo files(livestream.py) for open-gopro, there is a setting for
bitrate, resolution, field of view, etc.

Like I said, the qr code in labs dont seem to work. Even for next
livestream. It doesnt apply.


Le sam. 20 avr. 2024, 13 h 59, David Newman ***@***.***> a
écrit :

**Reply by dnewman-gpsw** [reply]
Next stream, same power on session.  Can you confirm it never works, as that would be a new bug.  Labs only, not a hybrid using Open GoPro and technically there is not official combination of Open GoPro and Labs.  Trying to fix that.

**Reply by maxdor1** [reply]
Hi,

I just tried. When my gopro boot, it says: feature activated BITL =8 . But when I start the stream with goprolabs qr code, the bitrate stays at 2500.. I tried it 3 times in a row and it stays at 2500. I rescan the BITL8 qr code, and the two next stream were also at 2500.

Next, I started a livestream with Quik, bitrate was 5000. Next I started another livestream with goprolabs qr code, bitrate was 5000.

QR code for BITL dont seems to work at al

**Reply by dnewman-gpsw** [reply]
Explain "I tried it 3 times in a row", if you rebooted it would be testing what I hoped.  Just do oMBITL=8 then !G for live, twice, do not reboot.

**Reply by maxdor1** [reply]
So I just tried what you said:

1. Boot the gopro 1st time, see uploaded image ![PXL_20240421_201459780](https://github.com/gopro/labs/assets/45298226/ed85097a-69a2-49bd-966d-fb0f550481b5)
2. Scan oMBITL=8
3. Scan !G
4. Check logs, bitrate is at 2500
5. Scan !G
6. Check logs, bitrate is still 2500

**Comment by maxdor1** [comment]
I'll get back to you later today.

Le sam. 20 avr. 2024, 14 h 23, David Newman ***@***.***> a
écrit :

**Comment by maxdor1** [comment]
I never reboot the Gopro between live session if thats what you mean.

Le sam. 20 avr. 2024, 21 h 03, David Newman ***@***.***> a
écrit :
